tests/libwnbd_tests/mock_wnbd_daemon.cc

```diff
@@ -234,6 +237,89 @@ void MockWnbdDaemon::Unmap(
         handler->MockStatus, NULL, 0);
 }
 
+void MockWnbdDaemon::PersistentReserveIn(
+    PWNBD_DISK Disk,
+    UINT64 RequestHandle,
+    UINT8 ServiceAction)
+{
+    MockWnbdDaemon* handler = nullptr;
+    ASSERT_FALSE(WnbdGetUserContext(Disk, (PVOID*)&handler));
+
+    WnbdRequestType RequestType = WnbdReqTypePersistResIn;
+    WNBD_IO_REQUEST WnbdReq = { 0 };
+    WnbdReq.RequestType = RequestType;
+    WnbdReq.RequestHandle = RequestHandle;
+    WnbdReq.Cmd.PersistResIn.ServiceAction = ServiceAction;
+    ASSERT_EQ(ServiceAction, RESERVATION_ACTION_READ_RESERVATIONS);
+
+    handler->ReqLog.AddEntry(WnbdReq);
+
+    switch (ServiceAction)
+    {
+    case RESERVATION_ACTION_READ_KEYS:
+    case RESERVATION_ACTION_READ_RESERVATIONS:
+        break;
+    default:
+        WnbdSetSense(
+            &handler->MockStatus,
+            SCSI_SENSE_ILLEGAL_REQUEST,
+            SCSI_ADSENSE_ILLEGAL_COMMAND);
+        break;
+    }
+
+    handler->SendIoResponse(
+        RequestHandle, RequestType,
+        handler->MockStatus, NULL, 0);
+}
+
+void MockWnbdDaemon::PersistentReserveOut(
+    PWNBD_DISK Disk,
+    UINT64 RequestHandle,
+    UINT8 ServiceAction,
+    UINT8 Scope,
+    UINT8 Type,
+    PVOID Buffer,
+    UINT32 ParameterListLength) 
+{
+    MockWnbdDaemon* handler = nullptr;
+    ASSERT_FALSE(WnbdGetUserContext(Disk, (PVOID*)&handler));
+
+    WnbdRequestType RequestType = WnbdReqTypePersistResOut;
+    WNBD_IO_REQUEST WnbdReq = { 0 };
+    WnbdReq.RequestType = RequestType;
+    WnbdReq.RequestHandle = RequestHandle;
+    WnbdReq.Cmd.PersistResOut.ServiceAction = ServiceAction;
+    ASSERT_EQ(ServiceAction, RESERVATION_ACTION_CLEAR);
+    WnbdReq.Cmd.PersistResOut.ParameterListLength = ParameterListLength;
+    WnbdReq.Cmd.PersistResOut.Scope = Scope;
+    ASSERT_EQ(Scope, RESERVATION_SCOPE_LU);
+    WnbdReq.Cmd.PersistResOut.Type = Type;
+    ASSERT_EQ(Type, RESERVATION_TYPE_EXCLUSIVE);
+
+    handler->ReqLog.AddEntry(WnbdReq);
+
+    switch (ServiceAction)
```

Comment:
It doesn't make much sense to have this switch statement here while having ``ASSERT_EQ(ServiceAction, RESERVATION_ACTION_CLEAR);`` a few lines above.